app/experimenter/experiments/serializers/design.py

```diff
@@ -198,32 +199,40 @@ def is_variant_valid(self, variants):
 
         return unique_names and non_empty
 
+    @transaction.atomic
     def update(self, instance, validated_data):
-        variants_data = validated_data.pop("variants", [])
-        instance = super().update(instance, validated_data)
+        try:
+            with transaction.atomic():
+                variants_data = validated_data.pop("variants", [])
+                instance = super().update(instance, validated_data)
+
+            if variants_data:
+                existing_variant_ids = set(
+                    instance.variants.all().values_list("id", flat=True)
+                )
+                # Create or update variants
+                for variant_data in variants_data:
+                    variant_data["experiment"] = instance
+                    variant_data["slug"] = slugify(variant_data["name"])
+                    ExperimentVariant(**variant_data).save()
+
+                # Delete removed variants
+                submitted_variant_ids = set(
+                    [v.get("id") for v in variants_data if v.get("id")]
+                )
+                removed_ids = existing_variant_ids - submitted_variant_ids
 
-        if variants_data:
-            existing_variant_ids = set(
-                instance.variants.all().values_list("id", flat=True)
-            )
-            # Create or update variants
-            for variant_data in variants_data:
-                variant_data["experiment"] = instance
-                variant_data["slug"] = slugify(variant_data["name"])
-                ExperimentVariant(**variant_data).save()
-
-            # Delete removed variants
-            submitted_variant_ids = set(
-                [v.get("id") for v in variants_data if v.get("id")]
-            )
-            removed_ids = existing_variant_ids - submitted_variant_ids
+                if removed_ids:
+                    ExperimentVariant.objects.filter(id__in=removed_ids).delete()
 
-            if removed_ids:
-                ExperimentVariant.objects.filter(id__in=removed_ids).delete()
+            self.update_changelog(instance, validated_data)
 
-        self.update_changelog(instance, validated_data)
+            return instance
+        except IntegrityError as e:
+            error_string = "Experimenter Error Occured: {}".format(e)
```

Comment:
Eh I would just make some pretty text like 'unable to save this change, please contact an experimenter administrator' or something.  Users don't know what db integrity errors mean.